internal/kafka/internal/handlers/admin_kafka_test.go

```diff
@@ -0,0 +1,389 @@
+						return dbapi.KafkaList{}, &api.PagingMeta{}, nil
+					},
+				},
+				accountService: account.NewMockAccountService(),
+			},
+			args: args{
+				url: "/kafkas",
+			},
+			wantStatusCode: http.StatusOK,
+		},
+		{
+			name: "should successfully return a non-empty kafkas list",
+			fields: fields{
+				kafkaService: &services.KafkaServiceMock{
+					ListFunc: func(ctx context.Context, listArgs *s.ListArguments) (dbapi.KafkaList, *api.PagingMeta, *errors.ServiceError) {
+						return dbapi.KafkaList{
+							{
+								Name: "test",
+							},
+						}, &api.PagingMeta{}, nil
+					},
+				},
+				accountService: account.NewMockAccountService(),
+			},
+			args: args{
+				url: "/kafkas",
+			},
+			wantStatusCode: http.StatusOK,
+		},
+		{
+			name: "should return an error if kafkaService List returns an error",
+			fields: fields{
+				kafkaService: &services.KafkaServiceMock{
+					ListFunc: func(ctx context.Context, listArgs *s.ListArguments) (dbapi.KafkaList, *api.PagingMeta, *errors.ServiceError) {
+						return nil, &api.PagingMeta{}, errors.GeneralError("test")
+					},
+				},
+			},
+			args: args{
+				url: "/kafkas",
+			},
+			wantStatusCode: http.StatusInternalServerError,
+		},
+		{
+			name: "should return an error if orderBy param is invalid",
+			args: args{
+				url: "/kafkas?orderBy=invalidField",
+			},
+			wantStatusCode: http.StatusBadRequest,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewAdminKafkaHandler(tt.fields.kafkaService, tt.fields.accountService, tt.fields.providerConfig)
+			req, rw := GetHandlerParams("GET", tt.args.url, nil)
+			h.List(rw, req)
+			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
+		})
+	}
+}
+
+func Test_Delete(t *testing.T) {
+	type fields struct {
+		kafkaService   services.KafkaService
+		accountService account.AccountService
+		providerConfig *config.ProviderConfig
+	}
+
+	type args struct {
+		url string
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantStatusCode int
+	}{
+		{
+			name: "should successfully accept kafka deletion request",
+			fields: fields{
+				kafkaService: &services.KafkaServiceMock{
+					RegisterKafkaDeprovisionJobFunc: func(ctx context.Context, id string) *errors.ServiceError {
+						return nil
+					},
+				},
+			},
+			args: args{
+				url: "/kafkas/{id}?async=true",
+			},
+			wantStatusCode: http.StatusAccepted,
+		},
+		{
+			name: "should return an error if async flag is not set to true when deleting kafka",
+			args: args{
+				url: "/kafkas/{id}",
+			},
+			wantStatusCode: http.StatusBadRequest,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewAdminKafkaHandler(tt.fields.kafkaService, tt.fields.accountService, tt.fields.providerConfig)
```

Comment:
`tt` is a range variable. It is reused on each iteration. If we run the tests in parallel the value of `tt` will be unpredictable. Better to copy `tt` and use the copy from inside the function.

Reply:
does it apply to all test cases? btw. I don't think we are running the tests in parallel

Reply:
do you think this should be a separate Jira to do it, including adding documentation?

Reply:
Yeah, I think so. 
I added a point to the tech meeting about adding some linters that will check for these errors.
When we will add the linter, we will have to fix all of these issues as well.

Reply:
ok. thanks. I will push the changes for the other comments shortly